site/src/lib/model/apiKey.model.ts

```diff
@@ -0,0 +1,239 @@
+import crypto from "crypto";
+import { Db, DBRef } from "mongodb";
+import { v4 as uuid } from "uuid";
+
+import { User } from "./user.model";
+
+type ApiKeyProperties = {
+  createdAt: Date;
+  displayName: string;
+  hashedString: string;
+  lastUsedAt?: Date | null;
+  publicId: string;
+  revokedAt?: Date | null;
+  salt: string;
+  useCount: number;
+  user: DBRef;
+};
+
+export type UserFacingApiKeyProperties = Omit<
+  ApiKeyProperties,
+  "hashedString" | "salt" | "user"
+>;
+
+type ApiKeyDocument = ApiKeyProperties;
+
+export class ApiKey {
+  createdAt: Date;
+  displayName: string;
+  hashedString: string;
+  lastUsedAt?: Date | null;
+  publicId: string;
+  revokedAt?: Date | null;
+  salt: string;
+  useCount: number;
+  user: DBRef;
+
+  static readonly COLLECTION_NAME = "bp-api-keys";
+
+  private static readonly HASHING_SECRET = process.env.HASHING_SECRET;
```

Comment:
```suggestion
  private static readonly HASHING_SECRET = mustGetEnvVar(process.env.HASHING_SECRET);
```

This will give us an early crash.

Reply:
We also won’t need these checks below:

```
    if (!this.HASHING_SECRET) {
      throw new Error(
        "Cannot hash API key: no HASHING_SECRET present in environment.",
      );
    }
```